subsys/bluetooth/services/ots/ots_oacp.c

```diff
@@ -699,6 +690,11 @@ ssize_t bt_gatt_ots_oacp_write(struct bt_conn *conn,
 		return BT_GATT_ERR(BT_ATT_ERR_INVALID_OFFSET);
 	}
 
+	if (k_work_is_pending(&ots->oacp_ind.work)) {
+		LOG_ERR("OACP Write received before indication sent");
+		return BT_GATT_ERR(BT_ATT_ERR_PROCEDURE_IN_PROGRESS);
+	}
+
```

Comment:
I recommend removing this check. It is not our job to police the remote. (That's the job of Bluetooth qualification.) And this if-statement can dangerously be mistaken to be a guarantee the notification has been sent.

Reply:
> I recommend removing this check. It is not our job to police the remote. (That's the job of Bluetooth qualification.) And this if-statement can dangerously be mistaken to be a guarantee the notification has been sent.

The spec does not say anything about this though. Write requests and indications are not related. The point of this check is that we only ever have 1 pending indication on our side, so we avoid implementing a queue